modules/core/src/main/java/org/apache/ignite/configuration/ClientConfiguration.java

```diff
@@ -115,18 +116,33 @@
     /** Reconnect throttling retries. See {@code reconnectThrottlingPeriod}. */
     private int reconnectThrottlingRetries = 3;
 
+    /**
+     * Try use other limited number of channels to send a request if default channel is not responding.
+     * 0 means try use all configured channels before fail.
+     */
+    private int channelsAttemptsLimit = 0;
```

Comment:
1. Do we really need this property? What is the use case when this is not `0`?
2. I think `retryLimit` is a better name

Reply:
I introduced this variable because number of configured nodes can dynamically changes. Earlier user could just configure addresses with limited number of nodes. With k8s configuration number of nodes are changes. And in case of large number of non-responding nodes (cluster is down, for example) without this limit client will wait for a long time for an exception.

Reply:
Rename to retryLimit